src/vdxf/classes/requestobjects/LoginRequestDetails.ts

```diff
@@ -0,0 +1,304 @@
+import { I_ADDR_VERSION } from '../../../constants/vdxf';
+import { fromBase58Check, toBase58Check } from "../../../utils/address";
+import varint from "../../../utils/varint";
+import { CompactIdAddressObject, CompactIdAddressObjectJson } from "../CompactIdAddressObject";
+
+export interface LoginRequestDetailsInterface {
+  version?: BigNumber;
+  flags?: BigNumber;  
+  requestId: string;
+  permissions?: Array<LoginPermission>;
+  callbackUri?: Array<CallbackUri>;
+  expiryTime?: BigNumber;
+}
+
+export interface LoginPermissionJson {
+  type: number;
+  identity: CompactIdAddressObjectJson;
+}
+
+export interface CallbackUriJson {
+  type: number;
+  uri: string;
+}
+export interface LoginPermission {
+  type: BigNumber;
+  identity: CompactIdAddressObject;
+}
+
+export interface CallbackUri {
+  type: BigNumber;
+  uri: string;
+}
+
+export interface LoginRequestDetailsJson {
+  version: number;
+  requestid: string;
+  flags: number;
+  permissions?: Array<LoginPermissionJson>;
+  callbackuri?: Array<CallbackUriJson>;
+  expirytime?: number;
+}
+
+export class LoginRequestDetails implements SerializableEntity {
+  version: BigNumber;
+  flags?: BigNumber;  
+  requestId: string;
+  permissions?: Array<LoginPermission>;
+  callbackUri?: Array<CallbackUri>;
+  expiryTime?: BigNumber;
+
+  // Version
+  static DEFAULT_VERSION = new BN(1, 10)
+  static VERSION_FIRSTVALID = new BN(1, 10)
+  static VERSION_LASTVALID = new BN(1, 10)
+
+  static FLAG_HAS_PERMISSIONS = new BN(1, 10);
+  static FLAG_HAS_CALLBACK_URI = new BN(2, 10);
+  static FLAG_HAS_EXPIRY_TIME = new BN(4, 10);
+
+  // Permission Types
+  static REQUIRED_ID = new BN(1, 10);
+  static REQUIRED_SYSTEM = new BN(2, 10);
+  static REQUIRED_PARENT = new BN(3, 10);
+
+  // Callback URI Types
+  static TYPE_WEBHOOK = new BN(1, 10);
+  static TYPE_REDIRECT = new BN(2, 10);
+  static TYPE_DEEPLINK = new BN(3, 10);
+
+  constructor(
+    request?: LoginRequestDetailsInterface 
+  ) {
+
+    this.version = request?.version || LoginRequestDetails.DEFAULT_VERSION;
+    this.requestId = request?.requestId || '';
+    this.flags = request?.flags || new BN(0, 10);
+    this.permissions = request?.permissions || null;
+    this.callbackUri = request?.callbackUri || null;
+    this.expiryTime = request?.expiryTime || null;
+  }
+
+  hasPermissions(): boolean {   
+      return this.flags.and(LoginRequestDetails.FLAG_HAS_PERMISSIONS).eq(LoginRequestDetails.FLAG_HAS_PERMISSIONS);
+  }
+
+  hasCallbackUri(): boolean {
+    return this.flags.and(LoginRequestDetails.FLAG_HAS_CALLBACK_URI).eq(LoginRequestDetails.FLAG_HAS_CALLBACK_URI);
+  }
+
+  hasExpiryTime(): boolean {
+    return this.flags.and(LoginRequestDetails.FLAG_HAS_EXPIRY_TIME).eq(LoginRequestDetails.FLAG_HAS_EXPIRY_TIME);
+  }
+
+  getByteLength(): number {
+    this.setFlags(); // Ensure flags are set correctly for length calculation
+    let length = 0;
+
+    length += varint.encodingLength(this.flags);
```

Comment:
flags should be varuint in all cases where we aren't trying to be compatible with the daemon. This would encode as compactSize (I've started doing this in the latest round of classes, older classes still use varint but varuint is a slight improvement).